### PR TITLE
fix: show error modal on all registration errors

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: Bugfix

This PR resolves [CX-1444]

### Description

Right now we show an error result screen for almost every error that happens in the registration flow in the **Register to bid** screen. 

There are two problems:
1. The error messages are not accurate (e.g. when the phone number is invalid the page shows a Network error)
2. When the user closes the error screen the app navigates back to the auction screen and the user has to click on "Register to bid" and enter all the data again.

This PR changes the error handling so that the user sees a simple error modal whenever an error happens.

| Before (generic) | Before (network) | After (generic) | After (phone) |
|---|---|---|---|
|image |image|image|image|

https://user-images.githubusercontent.com/4691889/119827262-3af1ee80-bef9-11eb-88fc-4ae289682cc7.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-1444]: https://artsyproduct.atlassian.net/browse/CX-1444

### CODE REVIEW

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ export class Registration extends React.Component<RegistrationProps, Registratio
       await this.createBidder()
     } catch (error) {
       if (!this.state.errorModalVisible) {
-        this.presentRegistrationError(error, RegistrationStatus.RegistrationStatusError)
+        this.presentErrorModal(error, null)
```
Comment (olerichter00) [inline src/lib/Components/Bidding/Screens/Registration.tsx:128]: Note: this presents an error modal with the default message.

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 2]
```diff
@@ -135,20 +135,22 @@ export class Registration extends React.Component<RegistrationProps, Registratio
...
-            this.presentErrorModal(errors, null)
+            this.presentErrorModal(errors, errorMessage)
             reject(errors)
           } else {
             done()
           }
         },
         onError: (error) => {
-          this.presentRegistrationError(error, RegistrationStatus.RegistrationStatusNetworkError)
+          this.presentErrorModal(error, errorMessage)
```
Comment (MounirDhahri) [inline src/lib/Components/Bidding/Screens/Registration.tsx:153]: **praise:** much better like this

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 3]
```diff
@@ -201,7 +203,7 @@ export class Registration extends React.Component<RegistrationProps, Registratio
             }
           }
         },
-        onError: (errors) => this.presentRegistrationError(errors, RegistrationStatus.RegistrationStatusNetworkError),
+        onError: (errors) => this.presentErrorModal(errors, null),
```
Comment (MounirDhahri) [inline src/lib/Components/Bidding/Screens/Registration.tsx:206]: too bad we don't get any information onError here

**Review by MounirDhahri — APPROVED** [review]

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
LGTM 🎉

**Review by brainbicycle — APPROVED** [review]
Nice! Thanks for doing this, this is much better UX. **thought:** is there any extra phone number validation we should be doing in the form to catch this before the user submits?

**Comment by MounirDhahri** [comment]
@brainbicycle good point. I think we can maybe use our existing phone input here as well

**Comment by olerichter00** [comment]
@MounirDhahri Good idea, I can create a ticket for it. Should be a #good-first-issue.

https://github.com/artsy/eigen/blob/90816d76ea263bfb790a061210a45fb16b1db64e/src/lib/Components/PhoneInput/PhoneInput.tsx#L102

**Comment by MounirDhahri** [comment]
That's the one, thank you for linking it @olerichter00. It can definitely a be a good first issue 🙌

File: src/lib/Components/Bidding/Screens/Registration.tsx [hunk 4]
```diff
@@ -275,12 +273,6 @@ export class Registration extends React.Component<RegistrationProps, Registratio
     }
   }
 
-  // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-  presentRegistrationError(error, status) {
-    console.error("Registration.tsx", error)
-    this.presentRegistrationResult(status)
-  }
-
```
Comment (pvinis) [inline src/lib/Components/Bidding/Screens/Registration.tsx:283]: One less thing to care about 👌👌

**Comment by pvinis** [comment]
Nice PR!

(Oops, instead of adding just a comment, I made it as a review. Oh well.
Also, I think you have the wrong video on the PR desc?)

**Pull request merged**